static/js/tools/choropleth/choropleth.tsx

```diff
@@ -576,6 +608,74 @@ class ChoroplethMap extends Component<PropsType, StateType> {
     this.mouseOutAction(index);
   };
 
+  /**
+   * Redirects the webclient to a particular geo. Handles breadcrumbs in redirect.
+   * @param {string} geoDcid to redirect to.
+   * @param {string} curGeo human-readable current geo, e.g. United States when geoDcid
+   *                 is country/USA.
+   */
+  private redirectToGeo = (geoDcid: string, curGeo: string): void => {
+    const url = new URL(window.location.href);
+    const currGeoId = url.searchParams.get("geoDcid");
+    let baseUrl = "/tools/choropleth";
+    baseUrl += buildChoroplethParams(["statVar", "pc", "mdom"]);
+    baseUrl += "&geoDcid=" + geoDcid;
+    baseUrl += "&bc=";
+
+    const idxOfCurr = this.state.breadCrumbs.findIndex(
+      (crumb) => crumb.geoId === geoDcid
+    );
+    let breadCrumbsList = this.state.breadCrumbs;
+    if (idxOfCurr == -1) {
+      breadCrumbsList.push({ geoId: currGeoId, geoName: curGeo });
+    } else {
+      breadCrumbsList = breadCrumbsList.slice(0, idxOfCurr);
+    }
+    this.setState({
+      breadCrumbs: breadCrumbsList,
+    });
+    for (let i = 0; i < breadCrumbsList.length; i++) {
+      baseUrl += i === 0 ? "" : ";";
+      baseUrl += breadCrumbsList[i].geoId + "~" + breadCrumbsList[i].geoName;
```

Comment:
would be good to use a constant for the separator (helps with readability and maintainability too)